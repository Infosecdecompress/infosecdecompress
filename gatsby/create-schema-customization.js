const createSchemaCustomization = ({ actions }) => {
  const { createTypes } = actions;
  const typeDefs = `
    type MarkdownRemark implements Node {
      frontmatter: Frontmatter
    }
    type SocialImage {
      publicURL: String
    }
    type Frontmatter {
      socialImage: SocialImage
    }
  `;

  createTypes(typeDefs);
};

module.exports = createSchemaCustomization;