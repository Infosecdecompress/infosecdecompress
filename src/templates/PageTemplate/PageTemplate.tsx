import React from "react";

import { graphql } from "gatsby";

import { Layout } from "@/components/Layout";
import { Page } from "@/components/Page";
import { Sidebar } from "@/components/Sidebar";
import { useSiteMetadata } from "@/hooks";
import { Node } from "@/types";

interface Props {
  data: {
    markdownRemark: Node;
  };
}

const PageTemplate: React.FC<Props> = ({ data }: Props) => {
  const { title: siteTitle, subtitle: siteSubtitle } = useSiteMetadata();
  const { html: body } = data.markdownRemark;
  const { frontmatter } = data.markdownRemark;
  const { title, description = "", socialImage } = frontmatter;
  const metaDescription = description || siteSubtitle;

  return (
    <Layout
      title={`${title} - ${siteTitle}`}
      description={metaDescription}
      socialImage={socialImage}
    >
      <Sidebar />
      <Page title={title}>
        <div dangerouslySetInnerHTML={{ __html: body }} />
      </Page>
    </Layout>
  );
};

export const query = graphql`
  query PageTemplate($slug: String!) {
    markdownRemark(fields: { slug: { eq: $slug } }) {
      id
      html
      frontmatter {
        title
        date
        description
        socialImage
      }
    }
  }
`;

export default PageTemplate;
