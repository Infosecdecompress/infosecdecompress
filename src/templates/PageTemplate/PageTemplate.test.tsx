import React from "react";
import renderer from "react-test-renderer";

import { StaticQuery, useStaticQuery } from "gatsby";

import * as mocks from "@/mocks";

import PageTemplate from "./PageTemplate";

const mockedStaticQuery = StaticQuery as jest.Mock;
const mockedUseStaticQuery = useStaticQuery as jest.Mock;

describe("PageTemplate", () => {
  const props = {
    data: {
      markdownRemark: mocks.markdownRemark,
    },
  };

  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(({ render }) =>
      render(mocks.siteMetadata),
    );
    mockedUseStaticQuery.mockReturnValue(mocks.siteMetadata);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<PageTemplate {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
