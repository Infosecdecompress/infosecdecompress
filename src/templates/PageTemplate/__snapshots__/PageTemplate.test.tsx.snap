// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTemplate renders correctly 1`] = `
<div>
  <div>
    <div>
      <div>
        <a
          href="/"
        />
        <div>
          <h2>
            <a
              href="/"
            >
              John Doe
            </a>
          </h2>
        </div>
        <p>
          InfosecDecompress.
        </p>
      </div>
      <nav>
        <ul>
          <li>
            <a
              href="/"
            >
              Articles
            </a>
          </li>
          <li>
            <a
              href="/pages/about"
            >
              About Me
            </a>
          </li>
          <li>
            <a
              href="/pages/contacts"
            >
              Contact Me
            </a>
          </li>
        </ul>
      </nav>
      <div
        style={
          {
            "margin": "5px auto",
          }
        }
      >
        <div
          style={
            {
              "margin": "0 auto",
            }
          }
        >
          <form
            onSubmit={[Function]}
          >
            <div
              style={
                {
                  "margin": "0 auto",
                }
              }
            >
              <input
                id="Search"
                onChange={[Function]}
                placeholder="搜尋文章"
                style={
                  {
                    "border": "1px solid #ccc",
                    "borderRadius": "4px",
                    "boxSizing": "border-box",
                    "display": "inline-block",
                    "margin": "0 auto",
                    "padding": "5px 5px",
                    "width": "250px",
                  }
                }
                value=""
              />
            </div>
          </form>
          <div>
            <table
              id="result"
              style={
                {
                  "border": "1px solid #ccc",
                  "borderCollapse": "collapse",
                  "borderRadius": "4px",
                  "display": "none",
                  "width": "250px",
                }
              }
            >
              <tbody />
            </table>
          </div>
        </div>
      </div>
      <div>
        Content is licensed under CC BY-SA 4.0
      </div>
    </div>
  </div>
  <div>
    <div>
      <h1>
        Perfecting the Art of Perfection
      </h1>
      <div>
        <div
          dangerouslySetInnerHTML={
            {
              "__html": "<p>An Essay on Typography by Eric Gill takes the reader back to the year 1930. The year when a conflict between two worlds came to its term. The machines of the industrial world finally took over the handicrafts.</p>",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
